### PR TITLE
NXP S32ZE update RTD 2.0.0

### PR DESCRIPTION
**Pull request by haduongquang**
This PR updates the NXP S32 bare-metal drivers and the tool-generated configurations to the Real-Time Drivers 2.0.0 for S32Z/E. Implement modifications the sources to integrate with Zephyr.

### CODE REVIEW

File: s32/drivers/s32ze/BaseNXP/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2022 NXP
+# Copyright 2022,2024 NXP
```
Comment (manuargue) [inline s32/drivers/s32ze/BaseNXP/CMakeLists.txt:1]: revert, no change in the file

**Review by manuargue — CHANGES_REQUESTED** [review]
pls follow NXP copyright notice format
> If there is a gap of years, use ", current year". (Note: use a space after a comma (,) but not around the dash (-).)

File: s32/drivers/s32ze/BaseNXP/CMakeLists.txt (hunk 1 above)
Comment (haduongquang) [reply s32/drivers/s32ze/BaseNXP/CMakeLists.txt]: I updated the copyright

File: s32/drivers/s32ze/BaseNXP/header/S32Z2.h [hunk 2]
```diff
@@ -1,237 +1,238 @@
...
-#include "S32Z2_USDHC.h"
-#include "S32Z2_VIRT_WRAP.h"
-#include "S32Z2_WATCHDOG.h"
-#include "S32Z2_XBIC.h"
-#include "S32Z2_XRDC.h"
-#include "S32Z2_SYSTICK.h"
-#include "S32Z2_MPU.h"
-
-#endif  /* #if !defined(S32Z2_H_) */
+/*
```
Comment (Dat-NguyenDuy) [inline s32/drivers/s32ze/BaseNXP/header/S32Z2.h:1]: What is happening with the line ending here, pls make sure all files have LF line break type

**Review by Dat-NguyenDuy — CHANGES_REQUESTED** [review]

File: s32/drivers/s32ze/Can_CANEXCEL/CMakeLists.txt [hunk 3]
```diff
@@ -4,4 +4,4 @@
 zephyr_include_directories(include)
 zephyr_library_sources(src/CanEXCEL_Ip_HwAccess.c)
 zephyr_library_sources(src/CanEXCEL_Ip_Irq.c)
-zephyr_library_sources(src/CanEXCEL_Ip.c)
+zephyr_library_sources(src/CanEXCEL_Ip.c)
```
Comment (manuargue) [inline s32/drivers/s32ze/Can_CANEXCEL/CMakeLists.txt:7]: do not introduce superfluous changes, last blank line must be kept. same applies for any other file.

**Review by manuargue — CHANGES_REQUESTED** [review]

File: s32/drivers/s32ze/BaseNXP/CMakeLists.txt (hunk 1 above)
Comment (manuargue) [reply s32/drivers/s32ze/BaseNXP/CMakeLists.txt]: copyright must be updated only if there was a (substantial) change in the file

Comment (haduongquang) [reply s32/drivers/s32ze/BaseNXP/CMakeLists.txt]: Exactly, I mean I reverted the copyright in the file that was not changed and updated the copyright as NXP copyright notice format in the changed files.

File: s32/drivers/s32ze/BaseNXP/header/S32Z2.h (hunk 2 above)
Comment (haduongquang) [reply s32/drivers/s32ze/BaseNXP/header/S32Z2.h]: I updated the EOL

File: s32/drivers/s32ze/Can_CANEXCEL/CMakeLists.txt (hunk 3 above)
Comment (haduongquang) [reply s32/drivers/s32ze/Can_CANEXCEL/CMakeLists.txt]: I reverted the last blank line

**Review by manuargue — APPROVED** [review]

**Review by Dat-NguyenDuy — APPROVED** [review]

**Comment by haduongquang** [comment]
Sorry for this inconvenience, I have updated the clock configuration for the RTU1 core to reach 1GHz, also have re-tested on this RTU

**Pull request merged**